src/foam/nanos/notification/NotificationAction.js

```diff
@@ -0,0 +1,22 @@
+foam.ENUM({
```

Comment:
This should be in a DAO, these will change as we add more are replace things. 
The notification settings themselves could register an ID/name to use elsewhere.

Reply:
@jlhughes I'm not sure what you mean here. This is an enum used by the Notification object to control whether the SendEmailNotificationDAO actually sends a notification. The confusion here comes by use having two ways of performing actions:

1. DAO.put
2. NotificationSetting.sendNotification (and the subclasses of NotificationSetting)

Since the NotificationSetting puts to the notificationDAO, there is duplicate logic when performing a user.doNotify call. Specifically, it calls the EmailSettings.sendNotification and also performs logic in the SendEmailNotificationDAO .. so unless there is a way to control whether a Notification instance should send an email (i.e. getEmailIsEnabled() call or the getAction() call), then we will have duplicate emails sent to the user.
I'm not sure how adding this to a DAO will help, or what it would even mean to have this is a DAO. Are you saying we should have Action types in a DAO and this is a reference to an action type .. and the NotificatSetting should have that same reference to the action type? I'm not sure why you would want this configurable since the *Setting object would have to be coded in order to enable it (e.g. SlackSetting) in which case the value could also be added to the enum. Unless you have a way to add NotificationSetting subtypes on the fly, I can't see why we would do this. 
I can add some logic here to match the enum instead of having it manually set per setting type.